### ISSUE TITLE
Best practice for cmake-conan and git submodules

### ISSUE THREAD

**Issue by LesleyLai** [conan-io/cmake-conan#203]
Hi, I have two projects A and B, where A is used as a submodule of B and gets included with `add_subdirectory`. Both projects are using cmake-conan and contain a `conanfile.txt`.

The problem is that both projects depend on some common libraries such as `fmt`. If I try to invoke cmake I would get a build time error:

```
add_library cannot create imported target "CONAN_PKG::fmt" because
  another target with the same name already exists.
```

What I tried:
1. Only list common packages in A, but that means
 - If A stop depend on that package, then B will break
 - I cannot have another submodule C that depends on that package
2. Guard the cmake-conan codes in A with `if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)`, but that means B needs to keep track of the dependencies used by A.

Any advice? I have a feeling that I am doing everything horribly wrong and there are better solutions to what I am doing.

**Comment by czoido** [comment]
Hi @LesleyLai,
I think that the way to go would be to create a conan package for the git submodule better than trying to add the submodule. Is that a possibility?

**Comment by LesleyLai** [comment]
@czoido Yeah, I like this idea. But also, I need to modify both projects simultaneously. The idea is that A served as a library shared across a bunch of projects. Does using submodule as Conan packages compatible with this kind of workflow?

**Comment by czoido** [comment]
If you want the share the A library for other projects across different machines I would recommend using a server for your libraries like [Artifactory CE](https://docs.conan.io/en/latest/uploading_packages/artifactory_ce.html) or [Conan Server](https://docs.conan.io/en/latest/uploading_packages/running_your_server.html) and to modify the library as you are developing you could make [your package editable](https://docs.conan.io/en/latest/developing_packages/editable_packages.html).
